### PR TITLE
rebuild cpp infer code

### PR DESCRIPTION
**Pull request by WenmuZhou**
1. keep the logic consistent with the whl package
2. cls support batch inference
3. support six run mode
    * det+cls+rec
    * det+rec
    * det
    * cls+rec
    * rec
    * cls

### CODE REVIEW

File: deploy/cpp_infer/include/ocr_cls.h [hunk 1]
```diff
@@ -52,14 +53,17 @@ class Classifier {
     this->cls_thresh = cls_thresh;
     this->use_tensorrt_ = use_tensorrt;
     this->precision_ = precision;
+    this->cls_batch_num_ = cls_batch_num;
 
     LoadModel(model_dir);
   }
+  double cls_thresh = 0.5;
```
Comment (littletomatodonkey) [inline deploy/cpp_infer/include/ocr_cls.h:60]: 改成0.9吧

File: test_tipc/configs/ch_ppocr_mobile_v2.0_det/model_linux_gpu_normal_normal_infer_cpp_linux_gpu_cpu.txt [hunk 2]
```diff
@@ -3,8 +3,8 @@ model_name:ocr_det
 use_opencv:True
 infer_model:./inference/ch_ppocr_mobile_v2.0_det_infer/
 infer_quant:False
-inference:./deploy/cpp_infer/build/ppocr det
---use_gpu:True|False
```
Comment (littletomatodonkey) [inline test_tipc/configs/ch_ppocr_mobile_v2.0_det/model_linux_gpu_normal_normal_infer_cpp_linux_gpu_cpu.txt:7]: 这里为啥把gpu去掉了

**Review by littletomatodonkey — APPROVED** [review]

File: deploy/cpp_infer/include/ocr_cls.h (hunk 1 above)
Comment (WenmuZhou) [reply deploy/cpp_infer/include/ocr_cls.h]: done

File: test_tipc/configs/ch_ppocr_mobile_v2.0_det/model_linux_gpu_normal_normal_infer_cpp_linux_gpu_cpu.txt (hunk 2 above)
Comment (WenmuZhou) [reply test_tipc/configs/ch_ppocr_mobile_v2.0_det/model_linux_gpu_normal_normal_infer_cpp_linux_gpu_cpu.txt:7]: done

**Review by littletomatodonkey — APPROVED** [review]

**Pull request merged**